src/shared/calendarspec.c

```diff
@@ -1196,15 +1196,18 @@ static int tm_within_bounds(struct tm *tm, bool utc) {
                 return negative_errno();
 
         /* Did any normalization take place? If so, it was out of bounds before */
-        bool good = t.tm_year == tm->tm_year &&
-                    t.tm_mon  == tm->tm_mon  &&
-                    t.tm_mday == tm->tm_mday &&
-                    t.tm_hour == tm->tm_hour &&
-                    t.tm_min  == tm->tm_min  &&
-                    t.tm_sec  == tm->tm_sec;
-        if (!good)
+        int cmp = CMP(t.tm_year, tm->tm_year) ?:
+                  CMP(t.tm_mon, tm->tm_mon) ?:
+                  CMP(t.tm_mday, tm->tm_mday) ?:
+                  CMP(t.tm_hour, tm->tm_hour) ?:
+                  CMP(t.tm_min, tm->tm_min) ?:
+                  CMP(t.tm_sec, tm->tm_sec);
```

Comment:
doesn't matter, but I'd split this out into a helper tm_cmp(), sounds generally useful

Reply:
Yeah, but it doesn't compare all fields... So I think this would be a potential pitfall to have `tm_cmp` that returns 0 when the structures are different.

Reply:
Hmm, if my grep-foo is correct, that's the only place where we do this kind of comparison on `struct tm`.  So I'll leave it as is for now.

Reply:
> Yeah, but it doesn't compare all fields... So I think this would be a potential pitfall to have tm_cmp that returns 0 when the structures are different.

The other fields are redundant though...